FiftyOne.Pipeline.CloudRequestEngine/FlowElements/CloudAspectEngineBase.cs

```diff
@@ -169,20 +169,33 @@ public override IList<IAspectPropertyMetaData> Properties
                     {
                         if (_aspectProperties == null)
                         {
-                            if (LoadAspectProperties(
-                                RequestEngine.GetInstance()) == false)
-                            {
-                                throw new PipelineException(string.Format(
-                                    CultureInfo.InvariantCulture, 
-                                    Messages.ExceptionFailedToLoadProperties,
-                                    ElementDataKey));
+                            var engineInstance = RequestEngine.GetInstance();
+                            try
+                            {
+                                if (LoadAspectProperties(engineInstance) == false)
+                                {
+                                    throw new PipelineException(string.Format(
+                                        CultureInfo.InvariantCulture,
+                                        Messages.ExceptionFailedToLoadProperties,
+                                        ElementDataKey));
+                                }
+                            }
+                            catch (CloudRequestException)
+                            {
+                                // ignore server errors, properties will be requested again
                             }
                         }
                     }
                 }
-                return _aspectProperties;
+                return _aspectProperties ?? new IAspectPropertyMetaData[0];
```

Comment:
The following call tree:
- [Pipeline.Pipeline](https://github.com/51Degrees/pipeline-dotnet/blob/main/FiftyOne.Pipeline.Core/FlowElements/Pipeline.cs#L242) (constructor)
  - [Pipeline.GetElementAvailableProperties](https://github.com/51Degrees/pipeline-dotnet/blob/main/FiftyOne.Pipeline.Core/FlowElements/Pipeline.cs#L497)
    - [Pipeline.AddAvailableProperties](https://github.com/51Degrees/pipeline-dotnet/blob/main/FiftyOne.Pipeline.Core/FlowElements/Pipeline.cs#L473)

contains a loop:
```
foreach (var property in element.Properties.Where(p => p.Available))
{
    if (availableElementProperties.ContainsKey(property.Name) == false)
    {
        availableElementProperties[property.Name] = property;
    }
}
```
^ that would crash if `element.Properties` is null

Thus, at the moment, we return a 0-length array from CloudAspectEngineBase instead of null when no meaningful result is available yet.

Maybe we should consider instead also lazifying [[Pipeline.ElementAvailableProperties](https://github.com/51Degrees/pipeline-dotnet/blob/main/FiftyOne.Pipeline.Core/FlowElements/Pipeline.cs#L196)] property in a way that prevents caching incomplete results.

Reply:
is it an option to add a null-check prior to that loop above to not crash? 
```
if (element.Properties != null) { 
 // loop
}
```
also will `AddAvailableProperties` be called again when `CloudRequestEngine` has already loaded the properties?

Reply:
also let's instead of creating a new object in that null-coalescing, maybe let's add a reference to static empty array

Reply:
the decision is to actually make `ElementAvailableProperties` lazy as well, otherwise some logic might be broken after the domain is back to life, but we never ask `CloudRequestEngine` for its properties again.

Reply:
Reverted adding try/catch to `CloudAspectEngineBase`.
Implemented requested changes to `Pipeline.ElementAvailableProperties`.